reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/SwitchIfEmpty.kt

```diff
@@ -0,0 +1,52 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import com.badoo.reaktive.utils.atomicreference.AtomicReference
+
+fun <T> Observable<T>.switchIfEmpty(otherObservable: Observable<T>): Observable<T> =
+    switchIfEmpty { otherObservable }
+
+fun <T> Observable<T>.switchIfEmpty(otherObservable: () -> Observable<T>): Observable<T> =
+    observableUnsafe { observer ->
+        val disposableWrapper = DisposableWrapper()
+        observer.onSubscribe(disposableWrapper)
+
+        subscribeSafe(
+            object : ObservableObserver<T>, ObservableCallbacks<T> by observer {
+                private val isEmpty = AtomicReference(true)
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onNext(value: T) {
+                    isEmpty.value = false
+                    observer.onNext(value)
+                }
+
+                override fun onComplete() {
+                    if (isEmpty.value) {
+                        otherObservable().subscribeSafe(
```

Comment:
You should try-catch client side callbacks (`otherObservable`). Something like this:
```
val nextObservable = 
    try {
        otherObservable()
    } catch (e: Throwable) {
        observer.onError(e)
        return
    }

nextObservable.subscribeSafe(
    // observer here
)
```

Also please add a test to cover the case when `otherObservable()` throws an exception.